### PR TITLE
Making the user stats reporter more robust to the intermittent errors…

### PR DESCRIPTION
**Pull request by cam156**
… returned by google analytics

### CODE REVIEW

File: sufia-models/lib/sufia/models/stats/user_stat_importer.rb [hunk 1]
```diff
@@ -47,6 +51,22 @@ def delay
       sleep @delay_secs
     end
 
+    def rescue_and_retry fail_message
```
Comment (mjgiarlo) [inline sufia-models/lib/sufia/models/stats/user_stat_importer.rb]: Include parens for consistency with the rest of the methods defined in this file?

File: sufia-models/lib/sufia/models/stats/user_stat_importer.rb [hunk 2]
```diff
@@ -47,6 +51,22 @@ def delay
...
+      retry_count = 0
+      begin
+        return yield
+      rescue Exception => e
```
Comment (mjgiarlo) [inline sufia-models/lib/sufia/models/stats/user_stat_importer.rb]: Rescue StandardError, not Exception: https://robots.thoughtbot.com/rescue-standarderror-not-exception

File: sufia-models/lib/sufia/models/stats/user_stat_importer.rb [hunk 3]
```diff
@@ -14,14 +15,17 @@ def import
       sorted_users.each do |user|
         start_date = date_since_last_cache(user)
 
+        # this user has already been processed today continue without delay
+        next if start_date.to_date == Date.today
```
Comment (mjgiarlo) [inline sufia-models/lib/sufia/models/stats/user_stat_importer.rb]: Do we have test coverage for this change?

File: sufia-models/lib/sufia/models/stats/user_stat_importer.rb (hunk 2 above)
Comment (cam156) [inline sufia-models/lib/sufia/models/stats/user_stat_importer.rb]: It is a StandardError, so I will change it.  Thanks!

File: sufia-models/lib/sufia/models/stats/user_stat_importer.rb [hunk 4]
```diff
@@ -7,6 +7,7 @@ def initialize(options={})
       @verbose = options[:verbose]
       @logging = options[:logging]
       @delay_secs = options[:delay_secs].to_f
+      @number_of_retries = options[:number_of_retries].to_f
```
Comment (grosscol) [inline sufia-models/lib/sufia/models/stats/user_stat_importer.rb]: Should the number of retries be an integer?

Comment (cam156) [inline sufia-models/lib/sufia/models/stats/user_stat_importer.rb]: Yep I will change it.

**Pull request merged**